src/UI/components/sources/source_form.tsx

```diff
@@ -0,0 +1,194 @@
+        />
+        <br />
+       
+      </div>
+      <br />
+      <div>
+        <Controller
+          name="article_title"
+          control={control}
+          render={({ field }) => (
+            <TextField
+              {...field}
+              type="text"
+              label="Article title:"
+              variant="outlined"
+            />
+          )}
+        />
+        <br />
+      </div>
+      <br />
+
+      <div>
+        <Controller
+          name="journal_title"
+          control={control}
+          render={({ field }) => (
+            <TextField
+              {...field}
+              type="text"
+              label="Journal title:"
+              variant="outlined"
+            />
+          )}
+        />
+        <br />
+      </div>
+      <br />
+
+      <div>
+        <Controller
+          name="year"
+          control={control}
+          render={({ field }) => (
+            <TextField
+              {...field}
+              type="number"
+              label="Year:"
+              variant="outlined"
+            />
+          )}
+        />
+        <br />
+      </div>
+      <br />
+
+      <div>
+        <Controller
+          name="report_type"
+          control={control}
+          render={({ field }) => (
+            <TextField
+              {...field}
+              type="text"
+              label="Report type:"
+              variant="outlined"
+            />
+          )}
+        />
+        <br />
+      </div>
+      <br />
+
+      <div>
+        <Controller
+          name="published"
+          control={control}
+          render={({ field }) => (
+            <>
+              <FormLabel>Published: </FormLabel>
+              <RadioGroup defaultValue="">
+                <FormControlLabel
+                  value="True"
+                  control={<Radio />}
+                  label="True"
+                ></FormControlLabel>
+                <FormControlLabel
+                  value="False"
+                  control={<Radio />}
+                  label="False"
+                ></FormControlLabel>
+              </RadioGroup>
+            </>
+          )}
+        />
+        <br />
+      </div>
+
+      <div>
+        <Controller
+          name="v_data"
+          control={control}
+          render={({ field }) => (
+            <>
+              <FormLabel>Vector Data: </FormLabel>
+              <RadioGroup defaultValue="">
+                <FormControlLabel
+                  value="True"
+                  control={<Radio />}
+                  label="True"
+                ></FormControlLabel>
+                <FormControlLabel
+                  value="False"
+                  control={<Radio />}
+                  label="False"
+                ></FormControlLabel>
+              </RadioGroup>
+            </>
+          )}
+        />
+      </div>
+      <br />
+
+      <Button variant="contained" size="large">
+        SUBMIT
+      </Button>
+    </form>
```

Comment:
When I try to submit a new source, I get a console error in the browser - `Warning: A component is changing an uncontrolled input to be controlled.` Do you see this too?